### PR TITLE
Premium content API + Grids video

### PR DESCRIPTION
**Pull request by undavide**
## Update the Premium content use case

- `isPremiumUser`, `startPremiumUpgradeIfFreeUser`, and `getPagesMetadata` APIs move from experimental to stable.
- Add simplified `startPremiumUpgradeIfFreeUser()` workflow
- Add a case with pre-export check for premium content
- Make linter happy
- Unrelated: add video to Grids tutorial

## Related Issue

See [this Slack post](https://photoshop.slack.com/archives/C03PFLPK1LM/p1708699830939549) from Kerri

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/guides/develop/use_cases.md [hunk 1]
```diff
@@ -100,47 +100,75 @@ document.getElementById("anchor").href = downloadUrl;
...
+      containsPremiumContent = true;
+      break;
     }
   }
+
+  if (containsPremiumContent && !userIsPremium) {
+    showPremiumContentError();
+    return;
+  } else {
+    const renditionOptions = {range: Range.entireDocument, format: RenditionFormat.png};
```
Comment (kerrishotts) [inline src/pages/guides/develop/use_cases.md:159]: This pattern requires the dev to have the export code in two places. We also lose the call to `createRenditions` which is is still required (before we can iterate over the `renditions` array).

I think it'd be good to pull out the export code into a separate function (that way error handling can be located in a single place)

I also just realized that if the user is in a co-editing scenario, they could still pass all the premium checks only to fail at the last minute when trying to create renditions, so we probably still need the try/catch section.

I think the following code would handle all the possible failure modes (but please double check me!)
```
import addOnUISdk from "https://new.express.adobe.com/static/add-on-sdk/sdk.js";
const { app, constants } = addOnUISdk;
const { ButtonType, Range, RenditionFormat, RenditionType, RenditionIntent } = constants;

const showPremiumContentError = async () => {
  const { buttonType } = await window.addOnUISdk.app.showModalDialog({
    variant: "error",
    title: "Export failed",
    description: "Sorry, we were not able to export your design. Some assets are only included in the Premium plan. Try replacing with something else or upgrading Adobe Express to a Premium plan.", 
    buttonLabels: { secondary: "Upgrade" }
  });

  if (buttonType === ButtonType.cancel) return false; // user is still not premium
  if (buttonType === ButtonType.secondary) {
    // Original flow (don't use anymore)
    // ❌ window.open("https://www.adobe.com/go/express_addons_pricing", "_blank")
    // 👇 Use startPremiumUpgradeIfFreeUser() instead 
    const hasUpgradedToPremium = await app.startPremiumUpgradeIfFreeUser();
	return hasUpgradedToPremium;
  }
}

const isRangeSafeToExport = await (range) => {
  const userIsPremium = await app.currentUser.isPremiumUser();
  const pages = await app.document.getPagesMetadata({range});
  const containsPremiumContent = pages.some(page => page.hasPremiumContent);
  return (containsPremiumContent && userIsPremium) || !containsPremiumContent;  
}

const exportDocument = async () => {
  // 👇 Testing purposes only!
  app.devFlags.simulateFreeUser = true; // Remove this line in production!

  let isSafeToExport = isRangeSafeToExport(Range.entireDocument);  
  if (!isSafeToExport) {
    const isNowPremiumUser = await showPremiumContentError();
    isSafeToExport = isNowPremiumUser;
  }
  
  if (isSafeToExport) {
    try {
      const renditions = await app.document.createRenditions({
        range: Range.entireDocument, format: RenditionFormat.png
      });
      renditions.forEach(rendition => { /* do your thing w/ the renditions */ });	  
    } catch (err) {
      // did someone just add premium content in the split second between
      // our original check? did the user just downgrade?
      if (err.message?.includes("USER_NOT_ENTITLED_TO_PREMIUM_CONTENT")) {
        return await exportDocument(); // try again
      }
    }
  }  
}

document.querySelector("#export").onclick = exportDocument;
```

**Review by kerrishotts — CHANGES_REQUESTED** [review]
Left a couple comments around the premium flow -- double check my code change suggestion (I was going off the top of my head) but I think this would handle all failure scenarios now.

File: src/pages/guides/develop/use_cases.md [hunk 2]
```diff
@@ -100,47 +100,75 @@ document.getElementById("anchor").href = downloadUrl;
...
 
-#### Option 1: Show premium content error with "Upgrade" option
+#### Option 1: Show a Premium Content error with the "Upgrade" option
 
-Display an error message when export/download fails due to the user not being entitled for premium content, and include a button to allow them to upgrade. *Be sure to update your `manifest.json` as outlined in the warning below the code snippet examples to allow the pricing page to properly load.*
+Display a warning message when the user is not entitled to export/download premium content, and include a button to allow them to upgrade. Please note that you can detect in advance if the user is entitled to premium content ([`isPremiumUser()`](../../references/addonsdk/app-currentUser.md#isPremiumUser)) and whether the page contains premium content ([`hasPremiumContent()`](../../references/addonsdk/app-document.md#pagemetadata)) in the first place. Try/catch blocks intercepting the `"USER_NOT_ENTITLED_TO_PREMIUM_CONTENT"` string in the error message are not recommended anymore.
```
[the remark this answers is not in the document]
Comment (undavide) [reply src/pages/guides/develop/use_cases.md:107]: Ooops! :-)

File: src/pages/guides/develop/use_cases.md (hunk 1 above)
Comment (undavide) [reply src/pages/guides/develop/use_cases.md:159]: Oh, I see what you're doing. It's very neat!!
I've fixed a couple of typos (an `await` that was meant to be `async` and a missing `await` when invoking `isRangeSafeToExport()`); other than that, it is perfect, thanks!

**Pull request merged**